lib/zuora/objects/payment_method_snapshot.rb

```diff
@@ -0,0 +1,98 @@
+module Zuora::Objects
```

Comment:
For the payment, yes. The payment has 2 objects associated with it: a pointer to the payment method in its current state and a snapshot of the payment method as it existed when the payment was processed/attempted.